packages/vm/src/imports.rs

```diff
@@ -215,41 +176,29 @@ pub fn do_scan<S: Storage + 'static, Q: Querier>(
         .try_into()
         .map_err(|_| CommunicationError::invalid_order(order))?;
 
-    let ffi_result = with_storage_from_context::<S, Q, _, _>(ctx, |store| {
+    let (result, gas_info) = with_storage_from_context::<S, Q, _, _>(ctx, |store| {
         Ok(store.range(start.as_deref(), end.as_deref(), order))
     })?;
-    let iterator_id = match ffi_result {
-        Ok((iterator, gas_info)) => {
-            process_gas_info::<S, Q>(ctx, gas_info)?;
+    process_gas_info::<S, Q>(ctx, gas_info)?;
+    let iterator_id = match result {
```

Comment:
Nitpic but this can be simplified further. Since the from cast is done on `?`, we can just focus on the positive case. Replace everything below `process_gas_info` with:

```rust
let iterator_id = add_iterator::<S, Q>(ctx, result?);
Ok(iterator_id)
```

Reply:
I usually don't like putting `?` inside function arguments but i agree with Ethan, this can be simplified as you describe :)
I'd maybe write it as:
```rust
let iterator = result?;
let iterator_id = add_iterator::<S, Q>(ctx, iterator);
Ok(iterator_id)
```

Reply:
Done in https://github.com/CosmWasm/cosmwasm/pull/489/commits/d2f188feed6c6f11756df15891e3ac5a39634443